src/main/java/com/raisetech/worldCup2022/controller/ParticipatingCountryController.java

```diff
@@ -37,21 +38,24 @@ public List<ParticipatingCountry> responseId(@PathVariable("id") int id) {
     }
 
     @PostMapping("/participating-country")
-    ResponseEntity<Map<String, String>> createParticipatingCountry(@RequestBody ParticipatingCountryForm form) {
-        URI url = UriComponentsBuilder.fromUriString("http://localhost:8080")
-                .path("/participating-country")
+    public ResponseEntity<Map<String, String>> createParticipatingCountry(@RequestBody ParticipatingCountryForm form, UriComponentsBuilder uriBuilder) {
+        ParticipatingCountry participatingCountry = participatingCountryService.createParticipatingCountry(form.getId(), form.getName(), form.getContinent());
+        URI url = uriBuilder
+                .path("/participating-country/" + participatingCountry.getId())
                 .build()
                 .toUri();
         return ResponseEntity.created(url).body(Map.of("message", "participating country was successfully created"));
     }
 
     @PatchMapping("/participating-country/{id}")
-    ResponseEntity<Map<String, String>> updateParticipatingCountry(@PathVariable("id") int id, @RequestBody ParticipatingCountryForm form) {
+    public ResponseEntity<Map<String, String>> updateParticipatingCountry(@PathVariable("id") int id, @RequestBody ParticipatingCountryUpdate update) {
+        participatingCountryService.updateParticipatingCountry(id, update.getName(), update.getContinent());
         return ResponseEntity.ok(Map.of("message", "successfully updated"));
     }
 
     @DeleteMapping("/participating-country/{id}")
-    public ResponseEntity<Void> deleteParticipatingCountry(@PathVariable("id") int delete) {
+    public ResponseEntity<Void> deleteParticipatingCountry(@PathVariable("id") int id, ParticipatingCountry delete) {
```

Comment:
idを指定すればレコード削除は可能なはず。
なのでControllerの引数はidのみでよいですね。
Service、Mapperも同様です。
```suggestion
    public ResponseEntity<Void> deleteParticipatingCountry(@PathVariable("id") int id) {
```